components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt

```diff
@@ -696,4 +697,50 @@ class WebExtensionSupportTest {
         store.waitUntilIdle()
         assertTrue(WebExtensionSupport.installedExtensions.isEmpty())
     }
+
+    @Test
+    fun `closes unsupported extension pages`() {
+        val store = spy(BrowserStore(BrowserState(
+            tabs = listOf(
+                createTab(id = "1", url = "https://www.mozilla.org"),
+                createTab(id = "2", url = "moz-extension://1234-5678/test"),
+                createTab(id = "3", url = "moz-extension://1234-5678-9/")
+            )
+        )))
+
+        val ext1: WebExtension = mock()
+        val ext1Meta: Metadata = mock()
+        whenever(ext1Meta.baseUrl).thenReturn("moz-extension://1234-5678/")
+        whenever(ext1.id).thenReturn("1")
+        whenever(ext1.url).thenReturn("url1")
+        whenever(ext1.getMetadata()).thenReturn(ext1Meta)
+        whenever(ext1.isEnabled()).thenReturn(true)
+        whenever(ext1.isAllowedInPrivateBrowsing()).thenReturn(true)
+
+        val ext2: WebExtension = mock()
+        whenever(ext2.id).thenReturn("2")
+        whenever(ext2.url).thenReturn("url2")
+        whenever(ext2.isEnabled()).thenReturn(true)
+        whenever(ext2.isAllowedInPrivateBrowsing()).thenReturn(false)
+
+        val engine: Engine = mock()
+        val callbackCaptor = argumentCaptor<((List<WebExtension>) -> Unit)>()
+        whenever(engine.listInstalledWebExtensions(callbackCaptor.capture(), any())).thenAnswer {
+            callbackCaptor.value.invoke(listOf(ext1, ext2))
+        }
+
+        WebExtensionSupport.initialize(engine, store)
+
+        store.waitUntilIdle()
+        assertNotNull(store.state.findTab("1"))
+        assertNotNull(store.state.findTab("2"))
+        assertNull(store.state.findTab("3"))
+
+        // Make sure we're running a single cleanup and stop the scope after
```

Comment:
@Amejia481 So, we do this in various places. I think the best way to verify that the scope is stopped is to apply the same change again, a change that would otherwise trigger the collect block, and make sure it has no effect.

This is what's happening here. I add another tab with an invalid extension URL, and make sure it isn't closed, because the scope is stopped. We don't want to keep it running once cleanup is successful, and don't need to.

Reply:
You can also comment out `scope.cancel` and you should see the test fail.

Reply:
Got thanks for the clarification 👍